library/test/src/test_result.rs

```diff
@@ -45,6 +45,18 @@ pub(crate) fn calc_result<'a>(
     time_opts: Option<&time::TestTimeOptions>,
     exec_time: Option<&time::TestExecTime>,
 ) -> TestResult {
+    let span_str = if desc.source_file != "" {
```

Comment:
Nit: what about using a name like?
```suggestion
    let at_location = if desc.source_file != "" {
```